onadata/libs/serializers/metadata_serializer.py

```diff
@@ -184,6 +184,10 @@ def validate(self, attrs):
                             _(u"User has no permission to "
                               "the dataview.")
                         })
+                    # ensure received xform contains submissions
```

Comment:
Why are we blocking a user from adding a media metadata if the XForm has no submissions ? This seems weird to me; In a scenario where a user has an external CSV won't this block them from uploading it ?

Reply:
My bad seems this is the linked dataset...?

Reply:
It should only restrict linked dataset.

Reply:
I think this applies to situation where we are using a linked media file. However, I do agree that we should not prevent a user from creating a link to a separate form due to no submissions on the form they are linking to.

Reply:
Also note, if the form being linked has submissions, it may not always be a guaranteed if we have submissions the user who downloads them will get any data to download. There is a case, where the additional restrictions on the form to be added as media requires that you only access records you are the submitter. In that situation, a user who had not made any submissions to the target form will have zero records.